### PR TITLE
OpenApi organize endpoints by tag

### PR DESCRIPTION
**Pull request by denis-yuen**
Working on #2982 

Seems to pass tests when running cli and ui2

Follow-up task, WTF is the following?
Screenshot from 2020-01-08 17-46-56

### CODE REVIEW

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 1]
```diff
@@ -2509,8 +2561,8 @@ components:
           - WDL
           - NFL
           - service
-          - cwl
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: @denis-yuen hmm, something to look into tommorow

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Coverage increased per file
===========================
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/dag/ElementsDefinition.java  100
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java  82
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java  2
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DAGHelper.java  100
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java  58
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java  16
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java  17
+ dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java  73
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java  20
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java  1
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryLabelHelper.java  5
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java  44
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java  82
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java  2
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java  11
+ dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java  34
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java  3
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java  88
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java  11
         

Coverage decreased per file
===========================
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/WorkflowPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/NotificationDAO.java  -85
- dockstore-webservice/src/main/java/io/swagger/api/impl/MetadataApiServiceImpl.java  -89
- dockstore-webservice/src/main/java/io/swagger/model/ToolDescriptor.java  -44
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java  -19
- dockstore-webservice/src/main/java/io/swagger/model/MetadataV1.java  -48
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  -37
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSHeader.java  -81
- dockstore-common/src/main/java/io/dockstore/common/Utilities.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/Limits.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java  -29
- dockstore-webservice/src/main/java/io/swagger/api/ToolClassesApiV1.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/SharedWorkflows.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ApiVersionConverter.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ExtendedUserData.java  -72
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Notification.java  -54
- dockstore-webservice/src/main/java/io/swagger/model/ToolDockerfile.java  -23
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/CollectionDAO.java  -59
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthorizer.java  -50
- dockstore-common/src/main/java/io/dockstore/common/OrganizationUpdateTime.java  -57
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java  -56
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/ServicePrototypePlugin.java  -64
- dockstore-webservice/src/main/java/io/swagger/model/ExtendedFileWrapper.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java  -20
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/Permission.java  -63
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Organization.java  -76
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java  -19
- dockstore-webservice/src/main/java/io/swagger/model/ToolVersionV1.java  -52
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/RSSListener.java  -30
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/RSSWorkflowPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/NotificationResource.java  -46
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenType.java  -7
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dockerhub/DockerHubTag.java  -15
- dockstore-webservice/src/main/java/io/swagger/model/ToolTestsV1.java  -24
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java  -93
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/CollectionOrganization.java  -100
- dockstore-common/src/main/java/io/dockstore/common/Repository.java  -56
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Service.java  -43
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolClassesApiServiceImpl.java  -45
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSFeed.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ToolDAO.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryStarredSerializer.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TokenDAO.java  -21
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/User.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java  -48
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Event.java  -22
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/MetadataResourceHelper.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/UserDAO.java  -29
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/OrganizationUser.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java  -18
- dockstore-common/src/main/java/io/dockstore/common/EntryUpdateTime.java  -55
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java  -9
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/Config.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java  -69
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ServiceDAO.java  -40
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java  -67
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java  -61
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Token.java  -9
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ImageRegistryFactory.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApi.java  -33
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java  -54
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DockstoreYaml.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSWriter.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java  -45
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZipSourceFileHelper.java  -65
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java  -18
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/StarrableResourceInterface.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicStateManager.java  -15
- dockstore-webservice/src/main/java/io/swagger/model/ToolV1.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsInterface.java  -79
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/StarRequest.java  -62
- dockstore-webservice/src/main/java/io/swagger/api/MetadataApiV1.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/ToolPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/TRSListener.java  -17
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/RSSToolPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java  -3
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java  -5
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java  -31
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/OrganizationDAO.java  -67
- dockstore-common/src/main/java/io/dockstore/common/Registry.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/BioWorkflowDAO.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/SitemapListener.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ManualRegistry.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java  -27
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java  -79
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  -34
- dockstore-webservice/src/main/java/io/swagger/api/ToolsApiV1.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedToolResource.java  -59
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ServiceResource.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java  -36
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSEntry.java  -79
         

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4786940&bid=15940853)

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml (hunk 1 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: Relates to 1.6.0 compatibility, should not be an issue

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public static final String CONTAINERS = "List and register entries in the dockstore (pairs of images + metadata (CWL and Dockerfile))";
+    public static final String CONTAINERTAGS = "List and modify tags for containers";
+    public static final String GA4GHV1 = "A curated subset of resources proposed as a common standard for tool repositories. Implements TRS [1.0.0](https://github.com/ga4gh/tool-registry-service-schemas/releases/tag/1.0.0) and is considered final (not subject to change)";
+    public static final String GA4GH = "A curated subset of resources proposed as a common standard for tool repositories. Implements TRS [2.0.0-beta.2](https://github.com/ga4gh/tool-registry-service-schemas/releases/tag/2.0.0-beta.2) . Integrators are welcome to use these endpoints but they are subject to change based on community input.";
+    public static final String EXTENDEDGA4GH = "Optional experimental extensions of the GA4GH API";
+    public static final String TOKENS = "List, modify, refresh, and delete tokens for external services";
+    public static final String WORKFLOWS = "List and register workflows in the dockstore (CWL or WDL)";
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java]: Missing Nextflow

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4786940&bid=15940853)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4786940&bid=15940853)

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+ *    Copyright 2019 OICR
```
Comment (agduncan94) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceConstants.java]: 2019 :scream:

**Pull request merged**